### ISSUE TITLE
Add strict requirements to upgrading chainID in 02-client UpgradeChain proposal

### ISSUE THREAD

**Issue by colin-axner** [cosmos/ibc-go#1309]
## Summary

Same as #1308, but it should be applied to the proposal handling in 02-client [here](https://github.com/cosmos/ibc-go/blob/18f1382d74fb447152dca13d95ac8a9fba7d8f83/modules/core/02-client/keeper/proposal.go#L81)

Same switch can be used:
```go
switch { 
case clienttypes.IsRevisionFormat(cs.ChainID):
    // check that the base of the cs.ChainID == base of upgradedClientState.ChainID
    oldChainIDBase := strings.TrimSuffix(chainID, fmt.Sprintf("-%d", clienttypes.ParseChainID(chainID)))
    newChainIDBase := chainID = strings.TrimSuffix(upgradedClientState.ChainID, fmt.Sprintf("-%d", clienttypes.ParseChainID(upgradedClientState.ChainID)))
    if oldChainIDBase != newChainIDBase {
        return err
    }

case cs.ChainID != upgradedClientState.ChainID
    // only allow upgrade from non-revision chainID to a chainID with "-0"
    if upgradedClientState.ChainID != clienttypes.SetRevisionNumber(cs.ChainID, 0) {
        return err
    }
}
```

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

**Comment by colin-axner** [comment]
Despite this [comment](https://github.com/cosmos/ibc-go/issues/1308#issuecomment-1112281283) we can still implement this proposed change. This will prevent chains from accidentally losing misbehaviour detection support. They will need to explicitly overwrite this functionality

**Comment by charleenfei** [comment]
after discussion with @colin-axner, we decided the best way forward for this issue is to close it for now.

the reasoning is because the implementation of this issue would basically cause an upgrade to fail if the revision format was not in the expected format. this is important because misbehaviour detection support will be disabled if a chain does not have conformant revision formats. however, the upgrade would fail only AFTER a proposal has passed, as this handler is run only after the passing of the proposal, which is bad UX. therefore, the issue should only be implemented alongside a complete documentation work packet of this aspect of 02 client.

in addition, the implementation of this issue would require casting the client to the Tendermint type, which would need to be updated with Tendermint v2. while this is possible in the short term, in combination with the UX issue mentioned above we decided it was best to postpone this for now until we can flesh out the entire work packet.

**Comment by gjermundgaraba** [comment]
This issue was mistakenly reopened during some integration work, so closing again now.
